File: malipo-php/src/Resources/Charges.php
<?php

namespace Malipo\Resources;

use Malipo\Malipo;

class Charges
{
    private $client;

    public function __construct(Malipo $client)
    {
        $this->client = $client;
    }

    public function create(array $params, array $options = [])
    {
        $headers = [];
        if (isset($options['idempotencyKey'])) {
            $headers['Idempotency-Key'] = $options['idempotencyKey'];
        }

        return $this->client->request('POST', '/charge', $params, $headers);
    }
}

File: malipo-flutter/lib/src/resources/charges.dart
import '../client.dart';
import '../models.dart';

class ChargesResource {
  final Malipo client;

  ChargesResource(this.client);

  Future<MalipoTransaction> create(ChargeCreateParams params, {String? idempotencyKey}) async {
    final headers = <String, String>{};
    if (idempotencyKey != null) {
      headers['Idempotency-Key'] = idempotencyKey;
    }

    final response = await client.request(
      'POST',
      '/charge',
      data: params.toJson(),
      headers: headers,
    );

    return MalipoTransaction.fromJson(response);
  }
}
